src/librustc_mir/transform/qualify_consts.rs

```diff
@@ -1038,32 +1038,28 @@ impl<'a, 'tcx> Checker<'a, 'tcx> {
 
         debug!("qualify_const: promotion_candidates={:?}", self.promotion_candidates);
         for candidate in &self.promotion_candidates {
-            match *candidate {
+            let promoted_place = match *candidate {
                 Candidate::Repeat(Location { block: bb, statement_index: stmt_idx }) => {
```

Comment:
It's a bug that this applies to `Repeat`. There is no way to make `Repeat` work by removing `StorageDead`.

Reply:
So we should treat `Repeat` like `Argument` here and do nothing right?

Reply:
That sounds correct.